packages/react-devtools-shared/src/backend/profilingHooks.js

```diff
@@ -214,6 +215,59 @@ export function createProfilingHooks({
     ((performanceTarget: any): Performance).clearMarks(markName);
   }
 
+  function beginMark(taskName: string, ending: string | number) {
+    // This name format is used in preprocessData.js so it cannot just be changed.
+    const startMarkName = `--${taskName}-start-${ending}`;
+    currentBeginMarksStack.push(startMarkName);
+    // This method won't be called unless these functions are defined, so we can skip the extra typeof check.
+    ((performanceTarget: any): Performance).mark(startMarkName);
+  }
+
+  function endMarkAndClear(taskName: string) {
+    const startMarkName = currentBeginMarksStack.pop();
+    if (!startMarkName) {
+      console.error(
+        'endMarkAndClear was unexpectedly called without a corresponding start mark',
+      );
+      return;
+    }
```

Comment:
Note to self: check if these can be safely stored in a stack, especially with concurrent root case